src/features/expenses/expenses.model.ts

```diff
@@ -1,26 +1,33 @@
-import mongoose, { Schema } from "mongoose";
+import mongoose, { Schema, Document } from "mongoose";
 import { Timestamp } from "../../db";
 
-export type Expense = {
+const dateFields = {
+  createdAt: { type: Date, default: Date.now, required: true },
+  updatedAt: { type: Date, default: Date.now, required: true },
+};
+
+export interface Expense extends Document, Timestamp {
   name: string;
   cause: string;
-  amount: number;
+  price: number;
   date: Date;
-} & Timestamp;
+  creator: mongoose.Types.ObjectId;
+}
 
-const expensesSchema = new Schema<Expense>(
+const expenseSchema = new Schema<Expense>(
   {
-    name: String,
-    amount: Number,
-    cause: String,
-    date: Date,
-    createdAt: Date,
-    updatedAt: Date,
+    name: { type: String, required: true },
+    price: { type: Number, required: true },
+    cause: { type: String, required: false },
+    date: { type: Date, required: false },
+    ...dateFields,
+    creator: {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "User",
+      required: false,
+    },
   },
   { versionKey: false, timestamps: true }
 );
 
-export const ExpansesModel = mongoose.model<Expense>(
-  "expenses",
-  expensesSchema
-);
+export const ExpensesModel = mongoose.model<Expense>("Expenses", expenseSchema);
```

Comment:
decide the convention

Reply:
by the documantion of mongoose its common to use CamelCase
so I left it as it is